test/Test.re

```diff
@@ -1,1061 +1,716 @@
-                       subtree: []
-                     },
-                     {
-                       id: 6,
-                       component: Component(ButtonWrapper.component),
-                       state: "",
-                       subtree: [
-                         {
-                           id: 7,
-                           component: Component(StatelessButton.component),
-                           state: "",
-                           subtree: [
-                             {
-                               component: Component(Div.component),
-                               state: "",
-                               id: 8,
-                               subtree: []
-                             }
-                           ]
-                         }
-                       ]
-                     }
-                   ]
-                 }
-               ]
-             }
-           ]
-         );
-      let rendered4 =
-        TestRenderer.update(
-          rendered3,
-          <ButtonWrapperWrapper wrappedText="updatedTextmodified" />
+        /* TODO Compare rendered0 and rendered1 */
+        /* check(
+             Alcotest.bool,
+             "Memoized nested box",
+             true,
+             ReasonReact.(
+               switch (rendered0, rendered1) {
+               | (
+                   IFlat([Instance({instanceSubTree: IFlat([x])})]),
+                   IFlat([
+                     Instance({instanceSubTree: INested(_, [_, IFlat([y])])})
+                   ])
+                 ) =>
+                 x === y
+               | _ => false
+               }
+             )
+           ); */
+      }
+    ),
+    (
+      "Test With Static Keys",
+      `Quick,
+      () => {
+        open ReasonReact;
+        GlobalState.reset();
+        let key1 = Key.create();
+        let key2 = Key.create();
+        let rendered0 =
+          RenderedElement.render(
+            ReasonReact.listToElement([
+              <Box key=key1 title="Box1unchanged" />,
+              <Box key=key2 title="Box2unchanged" />
+            ])
+          );
+        TestRenderer.convertElement(rendered0)
+        |> check(
+             renderedElement,
+             "Initial Boxes",
+             TestComponents.[
+               <Box id=1 state="Box1unchanged" />,
+               <Box id=2 state="Box2unchanged" />
+             ]
+           );
+        let (rendered1, _) as actual1 =
+          RenderedElement.update(
+            rendered0,
+            ReasonReact.listToElement([
+              <Box key=key2 title="Box2changed" />,
+              <Box key=key1 title="Box1changed" />
+            ])
+          );
+        assertUpdate(
+          ~label="Swap Boxes",
+          TestComponents.(
+            [
+              <Box id=2 state="Box2changed" />,
+              <Box id=1 state="Box1changed" />
+            ],
+            Some(
+              TestRenderer.{
+                subtreeChange: `Nested,
```

Comment:
What we've done in this test is reversing the order of boxes. It's not reflected in the subtreeChange... It'd cause lack of rerender/relayout which in turn would cause a visual bug. How will we solve this?